doc/UsersGuide/source/packagemanager.rst

```diff
@@ -1,70 +1,143 @@
+that contains basic models covering many fields of engineering. Other libraries, both open-source and
+commercial, are available to cover specific applications domains.
+
+When you start a simulation project using Modelica, it is common practice to collect all related system models
+in a project-specific package that you develop. The models in this package are often instantiated (e.g. by drag-and-drop
+in OMEdit) from released libraries, which are read-only for your project. This establishes a dependency between your
+project package and a certain version of a read-only package (or library), which is the one you have loaded in OMEdit
+and that you drag-and-drop components from.
+
+This dependency is automatically marked in your package by adding a `uses annotation
+<https://specification.modelica.org/maint/3.5/annotations.html#version-handling>`_ at the top level. For example, if you
+drag and drop components from MSL 4.0.0 into models of your package, the ``annotation(uses(Modelica(version="4.0.0")));``
+will be added automatically to it. This information allows OpenModelica to automatically load all the libraries
+that are required to compile the models in your own package next time you (or someone else, possibly on a different
+computer) loads your package, provided they are installed in places on the computer's file system where OpenModelica
+can find them.
+
+The default place where OpenModelica looks for packages is the so-called 
+`MODELICAPATH <https://specification.modelica.org/maint/3.5/packages.html#the-modelica-library-path-modelicapath>`_.
+You can check where it is by typing ``getModelicaPath()`` in the Interactive Environment. Installed
+read-only libraries are placed by default in the MODELICAPATH.
+
+When a new version of certain package comes out, `conversion annotations
+<https://specification.modelica.org/maint/3.5/annotations.html#version-handling>`_ in it declare whether your models using
+a certain older version of it can be used as they are with the new one, which is then 100% backwards compatible, or whether
+they need to be upgraded by running a conversion script, provided with the new version of the package. 
+
+If you install a new version of a library which is 100% backwards compatible with the previous ones, all your models that
+used the old one will automatically use the new one, which may contain bug-fixes, without the need of any further action.
+
+If the new version is not backwards compatible, instead, you will need to create a new version of
+your library that uses it, by running the provided conversion scripts. 
+
+OpenModelica has a package manager that can be used to install and update libraries on your computer, and is able to run
+conversion scripts. Keep in mind there are three stages in package usage: *available* packages are stored on the OSMC
+servers; *installed* packages are stored in the MODELICAPATH of your computer; *loaded* packages are loaded in memory
```

Comment:
*Nothing* is stored on our server except `index.json`. All zip-files are downloaded from github, gitlab, bitbucket, or simply URLs somewhere. (This makes licensing a non-issue for us since we are not distributing anything)

Reply:
We now say "*available* packages are indexed on the OSMC servers and can be downloaded from public repositories"